lms/djangoapps/instructor/views/data_access.py

```diff
@@ -0,0 +1,374 @@
+        query = Query(
+            query.query_type,
+            REVERSE_INCLUSION_MAP[query.inclusion],
+            course_id.make_usage_key(query.module_state_key.block_type, query.module_state_key.block_id),
+            query.filter_on,
+            query.entity_name,
+        )
+        make_single_query.apply_async(args=(course_id, query, group_id))
+
+
+def get_saved_queries(course_id, specific_group=None):
+    """
+    Get existing saved queries associated with a given course
+    """
+    if specific_group:
+        group = GroupedQuery.objects.filter(course_id=course_id, id=specific_group)
+    else:
+        group = GroupedQuery.objects.filter(course_id=course_id)
+    if len(group) == 0:
+        return ([], [], [])
+    relation = SubqueryForGroupedQuery.objects.filter(grouped__in=group)
+    queries = SavedQuery.objects.filter(id__in=relation.values_list(DatabaseFields.QUERY))
+    return (group, queries, relation)
+
+
+def get_temp_queries(course_id):
+    """
+    Get temporary queries associated with a course
+    """
+    queries_temp = TemporaryQuery.objects.filter(course_id=course_id)
+    return queries_temp
+
+
+@task(base=EmailWidgetTask)  # pylint: disable=not-callable
+def make_single_query(course_id, query, associate_group=None):
+    """
+    Make a single query for student information
+    """
+    temp_query = TemporaryQuery(
+        inclusion=INCLUSION_MAP.get(query.inclusion),
+        course_id=course_id,
+        module_state_key=query.entity_id,
+        filter_on=query.filter,
+        entity_name=query.entity_name,
+        query_type=query.query_type,
+        done=False,
+    )
+    temp_query.save()
+    try:
+        if query.query_type == QueryType.SECTION:
+            students = get_section_users(course_id, query)
+        else:
+            students = get_problem_users(course_id, query)
+        bulk_queries = []
+        for student_id, dummy0 in students:
+            row = StudentsForQuery(
+                query=temp_query,
+                inclusion=INCLUSION_MAP[query.inclusion],
+                student=User.objects.filter(id=student_id)[0],
+            )
+            bulk_queries.append(row)
+        StudentsForQuery.objects.bulk_create(bulk_queries)
+        TemporaryQuery.objects.filter(id=temp_query.id).update(done=True)  # pylint: disable=no-member
+        if associate_group is not None:
+            grouped_temp = GroupedTempQueryForSubquery(
+                grouped_id=associate_group,
+                query_id=temp_query.id,   # pylint: disable=no-member
+            )
+            grouped_temp.save()
+
+    except Exception as error:
+        TemporaryQuery.objects.filter(id=temp_query.id).update(done=None)  # pylint: disable=no-member
+        raise(error)
+
+    #on roughly every 10th query, purge the temporary queries
+    rand = random.random()
+    if rand > .9:
+        purge_temporary_queries()
+    return {temp_query.id: students}  # pylint: disable=no-member
```

Comment:
For all these tasks, is the return value important?  The issue here is that this must be encoded, and so testing on stage shows that all return values have to be json-encodeable (and possibly all task arguments, though we can toggle argument encoding to `pickle`, but I haven't found how to do that for return values.) (plus this is a hugely long list that might not be wise to serialize)

It might be better to program in a style where the tasks have no return values